packages/quantic/force-app/main/default/lwc/quanticSmartSnippetSource/quanticSmartSnippetSource.js

```diff
@@ -1,39 +1,61 @@
+import navigateToRecord from '@salesforce/label/c.quantic_NavigateToRecord';
+import opensInBrowserTab from '@salesforce/label/c.quantic_OpensInBrowserTab';
 import {LinkUtils} from 'c/quanticUtils';
+import {NavigationMixin} from 'lightning/navigation';
 import {LightningElement, api} from 'lwc';
 
+/** @typedef {import("coveo").Result} Result */
+
 /**
  * The `QuanticSmartSnippetSource` component displays the uri and the title of the smart snippet source.
  * @category Search
  * @example
- *  <c-quantic-smart-snippet-source uri={uri} title={title} actions={actions}></c-quantic-smart-snippet-source>
+ *  <c-quantic-smart-snippet-source actions={actions}></c-quantic-smart-snippet-source>
  * @internal
  */
-export default class QuanticSmartSnippetSource extends LightningElement {
+export default class QuanticSmartSnippetSource extends NavigationMixin(
+  LightningElement
+) {
   /**
-   * The click uri of the smart snippet source.
+   * The smart snippet source.
    * @api
-   * @type {string}
+   * @type {Result}
```

Comment:
It's not exactly this type though right?
We need to clearly be able to define what's expected in this complex object from the jsdoc.

Reply:
Something like this?

```
    /**
     * @typedef {Object} SourceRaw
     * @property {string} sfid Salesforce ID of the result
     * @property {string} sfkbid Salesforce ID of the Knowledge Article result
     * @property {string} sfkavid Salesforce ID of the Knowledge Article Version of the result
     */
...
   * @type {{title?: string, clickUri?: string, raw?: SourceRaw}}
```

Reply:
Actually it is exactly a `Result` object, in the Headless smart snippet state, it is of type `Result`.